### PR TITLE
Final PR Description

### PR DESCRIPTION
**Pull request by siddghosh108**
In this PR we have :-

- Chose the topic for your website and the API.
- Chose images, text, icons, fonts.
- Set up the repository and tools.
- Created the basic shared code on the home page.
- Started individual interfaces.
- Worked on individual interfaces.
- Completed the work on individual interfaces.
- Created a good README file.

image

Please give your review

Thanks
Link to the presentation Video:-

https://clipchamp.com/watch/949GDpXTmzL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,185 @@
...
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
+
+👤 **Siddhartha Ghosh**
+
+- GitHub: [@siddghosh108](https://github.com/siddghosh108)
+- Twitter: [@siddharthaghos9](https://twitter.com/siddharthaghos9)
+- LinkedIn: [@siddhartha-ghosh-65902718](https://www.linkedin.com/in/siddhartha-ghosh-65902718)
```
Comment (Efiamotu-1) [inline README.md:148]: - [ ] Seeing that it is a group project, I kindly advice that you add your partners profile as one of the authors of this project, that way he equally gets recognition for his contributions 👍

File: src/modules/counterComments.test.js
Comment (Efiamotu-1) [inline src/modules/counterComments.test.js:1]: - [ ] When writing test for the comment you are to make sure that you have tests for all edge cases. you are required to think at least about the scenario in which there are 0 items on the page (counter function should return 0) and write tests for that

File: src/modules/counterHomepage.test.js
Comment (Efiamotu-1) [inline src/modules/counterHomepage.test.js:1]: - [ ] When writing test for the number of meals on the home page you are to make sure that you have tests for all edge cases. you are required to think at least about the scenario in which there are 0 items on the page (counter function should return 0) and write tests for that

File: src/modules/counterHomepage.js
Comment (Efiamotu-1) [inline src/modules/counterHomepage.js:1]: - [ ] your counter function should cover all the edge cases you can think about. 👍

File: src/modules/counterComments.js
Comment (Efiamotu-1) [inline src/modules/counterComments.js:1]: - [ ] your counter function should cover all the edge cases you can think about. 👍

File: src/modules/renderMeals.js
Comment (Efiamotu-1) [inline src/modules/renderMeals.js:1]: - [ ] when i run your project, there seems to be an error which popup as shown in the image below, which is coming from this file, please kindly fix the error and resubmit your project  👍 
1er

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @siddghosh108 & @AbuTalha3, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Correct gitflow
- ✔️ Good job adding tv maze & involvement api
- ✔️ Good job adding tests 
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (siddghosh108) [reply README.md:148]: Includer my partner Abu Talha as an Author in README.md

👤 **Abu Talha**

- GitHub: [@githubhandle](https://github.com/AbuTalha3)
- Twitter: [@twitterhandle](https://twitter.com/AbuTalha8T)
- LinkedIn: [LinkedIn](https://www.linkedin.com/in/abu-talha-8203b252/)

File: src/modules/counterHomepage.test.js
Comment (siddghosh108) [reply src/modules/counterHomepage.test.js:1]: Added the following code segment to meet your requirement

test('should return 0 when there are no items on the homepage', () => {
    // Arrange: Provide an empty data array
    const data = [];
    // Act
    const result = counterHomepage(data);
    // Assert
    expect(result).toBe(0);
  });

File: src/modules/counterComments.js
Comment (siddghosh108) [reply src/modules/counterComments.js:1]: To meet your requirement put the following segment in the script
 it('should return 0 when there are no comments on the page', () => {
    // Arrange
    document.body.innerHTML = ''; // Empty the body
    // Act
    const list = document.querySelectorAll('.list-items');
    const result = counterComment(list);
    // Assert
    expect(result).toBe(0);
  });

File: src/modules/renderMeals.js
Comment (siddghosh108) [reply src/modules/renderMeals.js:1]: Thre is no more error, i have fixed it. Please see the image below

image

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- App runs locally without errors :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
